pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/AbstractRule.java

```diff
@@ -70,17 +78,14 @@ public void deepCopyValuesTo(AbstractRule otherRule) {
         otherRule.priority = priority;
         otherRule.propertyDescriptors = new ArrayList<>(getPropertyDescriptors());
         otherRule.propertyValuesByDescriptor = copyPropertyValues();
-        otherRule.ruleChainVisits = copyRuleChainVisits();
+        otherRule.ruleChainVisits = new LinkedHashSet<>(ruleChainVisits);
+        otherRule.classRuleChainVisits = new LinkedHashSet<>(classRuleChainVisits);
```

Comment:
Could be... but since we don't know exactly how the caller uses the deepCopy, we can't assume that they won't call `addRuleChainVisit`. If we copy `myStrategy`, then if it was not null, `buildTargetSelector` will never be called.

This is only relevant while `addRuleChainVisit` exists, when it's removed there will be no way to override the target selector of the implementation, and then it can be copied.